### PR TITLE
changes for branch prices

### PR DESCRIPTION
**Pull request by LNarendarReddy**
changes for branch prices

### CODE REVIEW

File: NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs [hunk 1]
```diff
@@ -24,7 +24,11 @@ public HttpResponseMessage getItemName(string itemCode)
                 var itemList = (from price in entities.ITEMPRICEs
                                 join code in entities.ITEMCODEs on price.ITEMCODEID equals code.ITEMCODEID
                                 join item in entities.ITEMs on code.ITEMID equals item.ITEMID
-                                where code.ITEMCODE1 == itemCode && code.DELETEDDATE == null && item.DELETEDDATE == null && price.DELETEDDATE == null
+                                where code.ITEMCODE1 == itemCode 
+                                            && code.DELETEDDATE == null 
+                                            && item.DELETEDDATE == null 
+                                            && price.DELETEDDATE == null
+                                            && price.BRANCHID != null
```
Comment (sainathgoud504) [inline NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs]: should be "branchid  == null"

File: NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs [hunk 2]
```diff
@@ -65,7 +69,10 @@ public HttpResponseMessage getPriceDetails(int itemCodeid)
             {
                 var itemList = (from price in entities.ITEMPRICEs
                                 join code in entities.ITEMCODEs on price.ITEMCODEID equals code.ITEMCODEID
-                                where price.ITEMCODEID == itemCodeid && price.DELETEDDATE == null && code.DELETEDDATE == null
+                                where price.ITEMCODEID == itemCodeid 
+                                            && price.DELETEDDATE == null 
+                                            && code.DELETEDDATE == null 
+                                            && price.BRANCHID != null
```
Comment (sainathgoud504) [inline NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs]: should be "branchid  == null"

File: NSRetail/NSRetail/Item/frmItemVisualize.cs [hunk 3]
```diff
@@ -43,12 +50,92 @@ private void gvItemPrice_FocusedRowChanged(object sender, DevExpress.XtraGrid.Vi
...
                 if (gvItemPrice.FocusedRowHandle < 0)
                     return;
-                gvStockSummary.Columns["ITEMCODE"].FilterInfo = 
+                gvStockSummary.Columns["ITEMCODE"].FilterInfo =
                     new ColumnFilterInfo($"[ITEMCODE] = '{gvItemPrice.GetFocusedRowCellValue("ITEMCODE")}'");
-                gcOffer.DataSource = 
+                gvBranchPrices.Columns["PARENTITEMPRICEID"].FilterInfo =
+                    new ColumnFilterInfo($"[PARENTITEMPRICEID] = '{gvItemPrice.GetFocusedRowCellValue("ITEMPRICEID")}'");
+                gcOffer.DataSource =
                     new ItemCodeRepository().GetOffers(gvItemPrice.GetFocusedRowCellValue("ITEMPRICEID"));
```
Comment (sainathgoud504) [inline NSRetail/NSRetail/Item/frmItemVisualize.cs]: use itemCodeRepository variable

File: Release/16122022/NSRetail_14122022.sql [hunk 4]
```diff
@@ -0,0 +1,1523 @@
...
+ALTER PROCEDURE USP_CU_BRANCHITEMPRICE
+@ITEMPRICEID INT = 0,
+@PARENTITEMPRICEID INT,
+@SALEPRICE DECIMAL(18,2),
+@BRANCHID INT,
+@USERID INT
+AS
+BEGIN
+	
+	IF EXISTS(SELECT 1 FROM ITEMPRICE WHERE ITEMPRICEID = @PARENTITEMPRICEID AND SALEPRICE = @SALEPRICE)
```
Comment (sainathgoud504) [inline Release/16122022/NSRetail_14122022.sql]: add and delete date is null

File: Release/16122022/NSRetail_14122022.sql [hunk 5]
```diff
@@ -0,0 +1,1523 @@
...
+	BEGIN
+		SELECT 'Saleprice already exists! Enter different value to create new one.'
+		RETURN;
+	END
+
+	IF NOT EXISTS(SELECT 1 FROM ITEMPRICE WHERE ITEMPRICEID = @ITEMPRICEID AND SALEPRICE = @SALEPRICE)
```
Comment (sainathgoud504) [inline Release/16122022/NSRetail_14122022.sql]: IF EXISTS(SELECT 1 FROM ITEMPRICE WHERE ITEMPRICEID = @ITEMPRICEID AND SALEPRICE <> @SALEPRICE 
 and deletedate is null)

File: Release/16122022/NSRetail_14122022.sql [hunk 6]
```diff
@@ -0,0 +1,1523 @@
...
+				WHERE (I.PARENTITEMID IS NULL OR ISNULL(I.ISOPENITEM, 0) = 1)     
+			) SDD ON SS.ITEMPRICEID = SDD.ITEMPRICEID
+	WHERE SS.BRANCHID = 45 AND (SDD.REASONID = 3 OR SDD.REASONID = 4)
+
+-- INSERTING SUPPLIER RETURNS RECORDS
+
+	INSERT INTO SUPPLIERRETURNS(SUPPLIERID,CREATEDBY,CREATEDDATE,STATUS)
+	SELECT DISTINCT DEALERID,@USERID,GETDATE(),0 FROM @dtbrd BRD
+	WHERE NOT EXISTS(SELECT 1 FROM SUPPLIERRETURNS SR
+	WHERE SR.SUPPLIERID = BRD.DEALERID AND ISNULL(SR.STATUS,0) = 0)
```
Comment (sainathgoud504) [inline Release/16122022/NSRetail_14122022.sql:449]: add delete date

File: Release/16122022/NSRetail_14122022.sql [hunk 7]
```diff
@@ -0,0 +1,1523 @@
...
+						GROUP BY IC.ITEMID
+					) BS ON BS.ITEMID = I.ITEMID
+				INNER JOIN	
+					(
+						SELECT IC.ITEMID, SUM(PBD.QUANTITY) AS SALEQUANTITY, SUM(PBD.WEIGHTINKGS) AS SALEWEIGHTINKGS
+						FROM 
+							POS_BILLDETAIL PBD
+							INNER JOIN POS_BILL PB ON PB.BILLID = PBD.BILLID AND PB.BRANCHCOUNTERID = PBD.BRANCHCOUNTERID
+							INNER JOIN BRANCHCOUNTER BC ON BC.COUNTERID = PB.BRANCHCOUNTERID
+							INNER JOIN ITEMPRICE IP ON IP.ITEMPRICEID = PBD.ITEMPRICEID
```
Comment (sainathgoud504) [inline Release/16122022/NSRetail_14122022.sql:1309]: add branchid is null case statement to pick parent price id

File: Release/16122022/NSRetail_Cloud_15122022.sql [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+  WHERE  IC.SYNCDATE > @SyncDate                                  
+  RETURN                                    
+ END                                    
+                                    
+ IF @EntityName = 'ITEMPRICE'                                    
+ BEGIN                                    
+  SELECT ITEMPRICEID, ITEMCODEID, SALEPRICE, MRP, GSTID, CREATEDDATE, UPDATEDATE, DELETEDDATE                                     
+  FROM ITEMPRICE IP                                    
+  WHERE IP.SYNCDATE > @SyncDate
+	AND (BRANCHID = NULL OR BRANCHID = @BranchID)
```
Comment (sainathgoud504) [inline Release/16122022/NSRetail_Cloud_15122022.sql]: branchid is null

File: NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs (hunk 1 above)
Comment (LNarendarReddy) [reply NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs]: Code changed

File: NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs (hunk 2 above)
Comment (LNarendarReddy) [reply NSRetail API/WebApiOauth2/Controllers/StockItemsController.cs]: Code Changed

File: NSRetail/NSRetail/Item/frmItemVisualize.cs (hunk 3 above)
Comment (LNarendarReddy) [reply NSRetail/NSRetail/Item/frmItemVisualize.cs]: Code Changed

File: Release/16122022/NSRetail_14122022.sql (hunk 4 above)
Comment (LNarendarReddy) [reply Release/16122022/NSRetail_14122022.sql]: Added condition

File: Release/16122022/NSRetail_14122022.sql (hunk 5 above)
Comment (LNarendarReddy) [reply Release/16122022/NSRetail_14122022.sql]: Changed this line

File: Release/16122022/NSRetail_14122022.sql (hunk 6 above)
Comment (LNarendarReddy) [reply Release/16122022/NSRetail_14122022.sql:449]: There is not Deleted Date for supplier returns

File: Release/16122022/NSRetail_Cloud_15122022.sql (hunk 8 above)
Comment (LNarendarReddy) [reply Release/16122022/NSRetail_Cloud_15122022.sql]: code Changed

**Review by sainathgoud504 — APPROVED** [review]

**Pull request merged**